appdynamics/client/template.go

```diff
@@ -0,0 +1,47 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+	"github.com/imroc/req"
+)
+
+func (c *AppDClient) createTemplateBaseUrl() string {
+	return fmt.Sprintf("%s/controller/restui/templates", c.BaseUrl)
+}
+
+func (c *AppDClient) createSetAssociatedDashboardsUrl(tierId int) string {
+	return fmt.Sprintf("%s/setAssociatedDashboards/%d?isTierDashboard=true", c.createTemplateBaseUrl(), tierId)
+}
+
+func (c *AppDClient) createGetAllDashboardTemplatesByTierUrl(tierId int) string {
+	return fmt.Sprintf("%s/getAllDashboardTemplatesByTier/%d?isTierDashboard=true", c.createTemplateBaseUrl(), tierId)
+}
+
+func (c *AppDClient) SetTemplateDashboardAssociations(tierId int, associations []int) (*Dashboard, error) {
+	resp, err := req.Post(c.createSetAssociatedDashboardsUrl(tierId), c.createAuthHeader(), req.BodyJSON(associations))
+	if resp.Response().StatusCode != 204 {
+		respString, _ := resp.ToString()
+		return nil, errors.New(fmt.Sprintf("Error creating association: %d, %s", resp.Response().StatusCode, respString))
+	}
+	if err != nil {
```

Comment:
You never return `*Dashboard` here. In any case you return nil. 

This function should return only `error` instead of `(*Dashboard, error)`